blog/26-04-2023-building-a-celo-based-decentralized-medical-record-system/building-a-celo-based-decentralized-medical-record-system.md

```diff
@@ -0,0 +1,833 @@
+            <div>
+              <label>
+                Name:
+                <input
+                  type="text"
+                  value={name}
+                  onChange={(e) => setName(e.target.value)}
+                  disabled={!!selectedRecordId}
+                />
+              </label>
+            </div>
+            <div>
+              <label>
+                Birthdate:
+                <input
+                  type="date"
+                  value={birthdate}
+                  onChange={(e) => setBirthdate(e.target.value)}
+                  disabled={!!selectedRecordId}
+                />
+              </label>
+            </div>
+            <div>
+              <label>
+                Medical Data:
+                <textarea
+                  value={medicalData}
+                  onChange={(e) => setMedicalData(e.target.value)}
+                />
+              </label>
+            </div>
+            {selectedRecordId ? (
+              <div>
+                <button type="submit">Update Record</button>
+                <button type="button" onClick={() => setSelectedRecordId(null)}>
+                  Cancel
+                </button>
+              </div>
+            ) : (
+              <button type="submit">Create Record</button>
+            )}
+          </form>
+          <div>
+            <h2>Your Medical Records</h2>
+            <MedicalRecordList records={records} onSelect={setSelectedRecordId} />
+          </div>
+          {error && <p>Error: {error}</p>}
+          {isLoading && <p>Loading...</p>}
+        </>
+      )}
+    </div>
+  );
+};
+export default MedicalRecord;
+```
+
+On running the command again:
+```bash
+npm start
+```
+
+You should see this browser window with the application.
+- Click on connect wallet.
+![image](https://user-images.githubusercontent.com/125604535/234740297-a9ff63e7-41c4-4215-8575-f0fd8e8f2a14.png)
+
+
+- Select the Metamask Wallet to connect to your Alfajores Account.
+![image](https://user-images.githubusercontent.com/125604535/234740845-40dd4588-96d9-4a96-b19b-fc6d55da1479.png)
+
+
+- Click on Confirm when your wallet loads up.
+![image](https://user-images.githubusercontent.com/125604535/234740917-72e15dcc-f733-44a2-8fe2-205049a62be6.png)
+
+- Fill in the form and click on Create Record.
+![image](https://user-images.githubusercontent.com/125604535/234740959-19a1e6aa-4203-4079-a508-cb985c7b0095.png)
+
+
+- Confirm this action in the metamask window pop-out.
+![image](https://user-images.githubusercontent.com/125604535/234741028-243237fa-149c-4843-8fae-77bbe155f47e.png)
+
+- Your record is now created and saved to the blockchain.
+![image](https://user-images.githubusercontent.com/125604535/234741082-93a71ac0-021a-42a9-8c2c-760ebd337da8.png)
+
+
+- You can view the details of the transaction on the Celo Explorer. Your Medical Records details will now be saved to the blockchain and can only be updated by an address with the  `DOCTOR_ROLE`.
+
+## Conclusion
```

Comment:
add a space before the first paragraph here

Reply:
done